### PR TITLE
Course details logic

### LINKED ISSUE #29: Course details instead of content
if i click on an course where i am not enrolled i should see the course details, not the item list

### PR DESCRIPTION
**Pull request by bjrne**
Fixes #29. The app now shows the course detail page if user is not enrolled. Upcoming course content can't be opened as well. In the future, the `accessible` attribute will be delivered via API.

### CODE REVIEW

File: xikolo-common/Data/Model/Course.swift [hunk 1]
```diff
@@ -58,6 +58,12 @@ class Course : BaseModel {
         return nil
     }
 
+    var accessible: Bool? {
```
Comment (invliD) [inline xikolo-common/Data/Model/Course.swift]: Please make this a non-optional Bool.

File: xikolo-ios/ViewControllers/Courses/DropdownViewController.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+//  DropdownViewController.swift
+//  xikolo-ios
+//
+//  Created by Bjarne Sievers on 29.10.16.
+//  Copyright © 2016 HPI. All rights reserved.
+//
+
+import UIKit
+
+class DropdownViewController: UIViewController {
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/DropdownViewController.swift]: What's this for, now? Why two dropdown VCs?

Comment (bjrne) [inline xikolo-ios/ViewControllers/Courses/DropdownViewController.swift]: I need the embedding ViewController for the popup on the iPad. The collapse to the small popup is not possible without this wrap and the additional layout constraints.

File: xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift [hunk 3]
```diff
@@ -28,12 +29,40 @@ class CourseDecisionViewController: UIViewController {
         NSNotificationCenter.defaultCenter().removeObserver(self);
     }
 
+    @IBAction func enroll(sender: UIBarButtonItem) {
+        if UserProfileHelper.isLoggedIn() {
+            UserProfileHelper.createEnrollement(course.id).onSuccess {
+                CourseHelper.refreshCourses().onSuccess(callback: { _ in
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: `onSuccess` should be called on the main thread anyway. Did you verify `dispatch_async` is necessary here?

File: xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift [hunk 4]
```diff
@@ -28,12 +29,40 @@ class CourseDecisionViewController: UIViewController {
...
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        updateContainerView(content)
+        decideContent()
         NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(switchViewController), name: NotificationKeys.dropdownCourseContentKey, object: nil)
     }
 
+    func decideContent() {
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: What about
```swift
        if(course.enrollment != nil) {
            navigationItem.rightBarButtonItem = nil
        }
        updateContainerView(course.accessible ? .learnings : .courseDetails)
```

File: xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift [hunk 5]
```diff
@@ -28,12 +29,40 @@ class CourseDecisionViewController: UIViewController {
         NSNotificationCenter.defaultCenter().removeObserver(self);
     }
 
+    @IBAction func enroll(sender: UIBarButtonItem) {
+        if UserProfileHelper.isLoggedIn() {
+            UserProfileHelper.createEnrollement(course.id).onSuccess {
```
Comment (bjrne) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: Is `Enrollement` a typo or is it intentional?

File: xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift [hunk 6]
```diff
@@ -28,12 +29,32 @@ class CourseDecisionViewController: UIViewController {
...
+                CourseHelper.refreshCourses().onSuccess(callback: { _ in
+                    self.updateContainerView(.learnings)
+                })
+            }
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: There are nested Futures here. Let's do
```swift

            UserProfileHelper.createEnrollement(course.id)
                .flatMap { CourseHelper.refreshCourses() }
                .onSuccess { _ in
                    self.updateContainerView(.learnings)
                }
```

**Comment by bjrne** [comment]
This will be refactored slightly to use the new api attributes after #100 is merged.

**Comment by jprberlin** [comment]
#100 is merged

**Pull request merged**